Graphics/Player judgment.lua

```diff
@@ -45,10 +45,27 @@ return Def.ActorFrame{
 		local frame = TNSFrames[ param.TapNoteScore ]
 		if not frame then return end
 
-		-- most judgment sprite sheets have 12 frames; 6 for early judgments, 6 for late judgments
+		-- If the judgment font contains a graphic for the additional white fantastic window...
+		if sprite:GetNumStates() == 7 or sprite:GetNumStates() == 14 then
+			if ToEnumShortString(param.TapNoteScore) == "W1" then
+				if mods.ShowFaPlusWindow then
+					-- If this W1 judgment fell outside of the FA+ window, show the white window
+					if not IsW0Judgment(param, player) then
+						frame = 1
+					end
+				end
+				-- We don't need to adjust the top window otherwise.
+			else
+				-- Everything outside of W1 needs to be shifted down a row.
+				frame = frame + 1
+			end
+		end
+
+
+		-- most judgment sprite sheets have 12 or 14 frames; 6/7 for early judgments, 6/7 for late judgments

```

Comment:
Are there actually any sprite sheets with 6 ot 7 frames? I have never seen that.
If you haven't seen it either we can just remove support for them and the code will be easier to follow.

Reply:
I actually have seen many 1x6 and 1x7 sprite sheets. We just don't include them in SL.